llvm/include/llvm/TargetParser/Triple.h

```diff
@@ -1109,6 +1109,11 @@ class Triple {
            Env == llvm::Triple::EABIHF;
   }
 
+  /// Tests if the target represents a GPU which can be offloaded to.
+  bool isOffloadingTargetGPU() const {
```

Comment:
Is it better to use `isOffloadingTarget`?

Reply:
sure, thats good point. i think doing that will resolve your other comment about SPIR-V being a gpu. ill implement this.